src/entry.ts

```diff
@@ -160,13 +161,24 @@ class Entry {
     } else if (is.string(this.data)) {
       entry.textPayload = this.data;
     }
+    // Format log timestamp
     if (is.date(entry.timestamp)) {
       const seconds = (entry.timestamp as Date).getTime() / 1000;
       const secondsRounded = Math.floor(seconds);
       entry.timestamp = {
         seconds: secondsRounded,
         nanos: Math.floor((seconds - secondsRounded) * 1e9),
       };
+    } else if (is.string(entry.timestamp)) {
+      // Convert RFC3339 "Zulu" timestamp into a format that can be parsed to Date
+      const zuluTime = entry.timestamp as string;
+      const ms = Date.parse(zuluTime.split(/[.,Z]/)[0] + 'Z');
+      const reNano = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.(\d{0,9})Z$/;
+      const nanoSecs = zuluTime.match(reNano)?.[1];
+      entry.timestamp = {
+        seconds: ms ? Math.floor(ms / 1000) : 0,
+        nanos: nanoSecs ? parseInt(nanoSecs.padEnd(9, '0')) : 0,
```

Comment:
It's generally safer to use the `Number` constructor instead of `parseInt`.  `parseInt` is more liberal in what it accepts, and in some cases can make assumptions about the radix (leading 0s and octal numbers, etc).